### PR TITLE
PullRequest for Issue16: Add enableFlattening for the server api and client UI

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
Addd enableFlattening feature for AMDSClientDataRequest to ask AMDS server to flatten the data for client

https://github.com/acquaman/AcquamanDataServer/issues/16

### CODE REVIEW

File: source/Connection/AMDSClientTCPSocket.cpp [hunk 1]
```diff
@@ -56,27 +55,26 @@ void AMDSClientTCPSocket::readFortune()
...
+		} else {
+			// finish reading this message, waiting for the future data
+			return;
 		}
 	}
 
-	// finish reading this message, waiting for the future data
-	if (waitingMorePackages_)
-		return;
-
```
Comment (XuetaoLiu-S) [inline source/Connection/AMDSClientTCPSocket.cpp:79]: This change imported a bug. It is fixed in Issue 17

**Comment by iainworkman** [comment]
Looks good.

File: source/AMDSClient.h [hunk 2]
```diff
@@ -56,6 +56,7 @@ private slots:
 	QSpinBox* amptekIndex;
 	QComboBox *bufferNameComboBox_;
 	QCheckBox* includeStatusDataCheckbox;
```
Comment (dretrex) [inline source/AMDSClient.h:60]: All of these internal variables should have a trailing _ to ensure that they don't have conflicts with other names.

**Comment by dretrex** [comment]
Otherwise looks good.
Time Estimate: 5 minutes
Time estimate, @iainworkman?

File: source/AMDSClient.h (hunk 2 above)
Comment (XuetaoLiu-S) [inline source/AMDSClient.h:60]: This file is replaced with AMClientUi.h in a later PR and the comments will be applied there.

**Pull request merged**